### ISSUE TITLE
Passing unserializable values into action logger causes system to hang

### ISSUE THREAD

**Issue by Billy-** [storybooks/storybook#1839]
Title pretty much says it all. Kinda related to #132, you just need to check if the object is serializable rather than only checking if it's a SyntheticEvent type

**Comment by Billy-** [comment]
Of course, the obvious workaround is to pull out / modify the values:

```
const logOnChange = action('onChange')
...
    onChange={(foo, unserializable) => logOnChange(foo)}
...
```

**Comment by danielduan** [comment]
You're more than welcome to open a PR to fix the issue

**Comment by Billy-** [comment]
If it were as simple as wrapping `serialize` in a `try {} catch` then I would. However as mentioned it causes the system to hang -- and no errors are output at the end of it. So I'm not really sure where to go from there.

**Comment by dangreenisrael** [comment]
@Billy - Could you post detailed instructions on how to replicate this issue.

**Comment by danielduan** [comment]
I think it has to do with how our addon channels serialize the communication so it's probably not just the addon-actions.

something like this should cause it to hang when it should be dropping the object key:
```
const a = {
  b: this
}
```

**Comment by Billy-** [comment]
@dangreenisrael I have come across this issue while using `react-flatpick`. Just throw a quick story together for it and add an action hander to the onChange:
```
...
onChange={action('onChange')}
...
```
This causes the window to freeze up for around 20-30 seconds and then resumes and nothing happens

**Comment by danielduan** [comment]
@Billy- might be related to this? https://github.com/coderhaoxin/react-flatpickr/issues/43

**Comment by Billy-** [comment]
I don't think so. As mentioned above, this issue can be worked around by using an arrow function and pulling out only the variables you want. It is only if you pass the action logger straight into the onChange that it happens - because the second parameter is unserializable

**Comment by dangreenisrael** [comment]
@Billy-  Could you post a new repo reproducing this problem without the use of 'react-flatpikr'?

**Comment by Billy-** [comment]
I don't really know the exact cause of the bug, so I'd have to put some time into actually trying to track that down. Unfortunately I don't really have the time to contribute that. Sincerest apologies!

**Comment by Hypnosphi** [comment]
Fixed in `3.3.0-alpha.4`
